### DISCUSSION TITLE
REST API services authorizaton via oauth2 Github provider

### DISCUSSION THREAD

**Discussion by pavolo99** [micronaut-projects/micronaut-security#832] [Q&A]
Hi, I am trying to implement an authentication via Github provider.
But after retrieving access token from github, I am not able to get authenticated access token.

```
@Secured(SecurityRule.IS_AUTHENTICATED) 
@View("repos") 
@Get 
Map<String, Object> index(Authentication authentication) {
     List<GithubRepo> repos = githubApiClient.repos(CREATED, DESC, authorizationValue(authentication));  
      Map<String, Object> model = new HashMap<>();
      model.put(REPOS, repos);
      return model;
}
```

Authentication is accessible only from the thymleaf view.

I need to find a way to secure REST API services via authorization header. 
Thank for response.

**Comment by sdelamo** [comment]
We have an [Secure a Micronaut application with Github](https://guides.micronaut.io/latest/micronaut-oauth2-github.html). Have you checked that guide? 

When we generate a new JWT, we allow you to get the orginal access token and idtoken as part of the authentication attributes. 

> I need to find a way to secure REST API services via authorization header.

Not sure I understand this question.
